ibm/service/database/validators.go

```diff
@@ -0,0 +1,137 @@
+// Copyright IBM Corp. 2017, 2022 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package database
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/IBM/cloud-databases-go-sdk/clouddatabasesv5"
+)
+
+/* TODO move other validators in here */
+
+/* VERSION VALIDATOR */
+
+type Version struct {
+	Version     string
+	Type        string
+	Status      string
+	IsPreferred bool
+	Transitions []VersionTransition
+}
+
+type VersionTransition struct {
+	Application         string
+	Method              string
+	FromVersion         string
+	ToVersion           string
+	SkipBackupSupported *bool
+}
+
+func expandVersion(version clouddatabasesv5.VersionsCapabilityItem) *Version {
+	expandedVersion := &Version{
+		Version:     *version.Version,
+		Status:      *version.Status,
+		IsPreferred: *version.IsPreferred,
+	}
+
+	if version.Transitions != nil {
+		var transitions []VersionTransition
+
+		for _, transition := range version.Transitions {
+			versionTransition := VersionTransition{
+				Application: *transition.Application,
+				Method:      *transition.Method,
+				FromVersion: *transition.FromVersion,
+				ToVersion:   *transition.ToVersion,
+			}
+
+			if skipBackup := transition.SkipBackupSupported; skipBackup != nil {
+				versionTransition.SkipBackupSupported = skipBackup
+			}
+
+			transitions = append(transitions, versionTransition)
+		}
+		expandedVersion.Transitions = transitions
+	}
+
+	return expandedVersion
+}
+
+func (v *Version) isVersionUpgradeAllowed(version string) bool {
+	for _, transition := range v.Transitions {
+		if transition.ToVersion == version && transition.Method == "in-place" {
+			return true
+		}
+	}
+	return false
+}
+
+func (v *Version) isSkipBackupUpgradeAllowed(version string) bool {
+	for _, transition := range v.Transitions {
+		if transition.ToVersion == version && transition.SkipBackupSupported != nil && transition.Method == "in-place" {
+			return *transition.SkipBackupSupported
+		}
+	}
+	return false
+}
+
+func (v *Version) hasUpgradeVersions() bool {
+	for _, transition := range v.Transitions {
+		if transition.Method == "in-place" {
+			return true
+		}
+	}
+	return false
+}
+
+func (v *Version) getAllowedVersionsList() []string {
+	var allowedList []string
+	for _, transition := range v.Transitions {
+		if transition.Method == "in-place" {
+			allowedList = append(allowedList, transition.ToVersion)
+		}
+	}
+	return allowedList
+}
+
+var fetchDeploymentVersionFn = fetchDeploymentVersion
+
+func fetchDeploymentVersion(instanceId string, meta interface{}) *Version {
+	capability, err := getDeploymentCapability("versions", instanceId, "classic", "us-south", meta)
+	if err != nil {
+		log.Fatalf("Error fetching deployment versions: %v", err)
+	}
+
+	if capability == nil || capability.Versions == nil || len(capability.Versions) == 0 {
+		return nil
+	}
+
+	// Expand the first version as there's only one expected
+	version := expandVersion(capability.Versions[0])
```

Comment:
> Expand the first version as there's only one expected

Only one is expected because it is getting the currently deployed version?

Reply:
Yes

Reply:
@obai-1 pushed an update for location to be passed in